common/djangoapps/student/admin.py

```diff
@@ -183,6 +178,43 @@ class CourseEnrollmentAdmin(admin.ModelAdmin):
     def queryset(self, request):
         return super(CourseEnrollmentAdmin, self).queryset(request).select_related('user')
 
+    def has_permission(self, request, method):
+        """
+        Returns True if the given method is allowed.
+
+        Access to these admin views is restricted because it makes DB quries that impact performance enough to cause a
+        site outage, cf https://openedx.atlassian.net/browse/OPS-2943
+
+        Enable these views using the waffle switch: `student.coursenrollment.admin`
+        """
+        if waffle.switch_is_active('student.coursenrollment.admin'):
```

Comment:
1. Please use waffle_utils. Here is [an example use in the student app.](https://github.com/edx/edx-platform/blob/df52ff00382ba34de8a69621eac8651ff44d4874/common/djangoapps/student/views/dashboard.py#L664).
2. I'd declare a constant like `COURSE_ENROLLMENT_ADMIN_SWITCH` near the top.
3. You could add a `STUDENT_WAFFLE_NAMESPACE = 'student'` to student's __init__.py, but that is up to you.
4. Could you move the comment about OPS-2943 to above the switch declaration?  Also, we might want to reword a little:
```
# This switch exists because these admin views make DB queries that impact performance.
# In a large enough deployment of Open edX, this is enough to cause a  site outage.
# See https://openedx.atlassian.net/browse/OPS-2943
```

Reply:
Oo thanks for the pointers @robrap ! -- I had forgotten that we're caching waffle flags and switches with `waffle_utils`.

Have followed your recommendations with 0b3231e .

Reply:
Thanks for the switch (pun intended). 😉 Hopefully one day we’ll have additional reporting features on these as well.

Reply:
No worries ;)

Reporting would be great.  Also would love a nice Admin GUIs which shows you all the switches/flags available and lets you tweak them easily, without having to dig through code to find the right namespace and name!

Reply:
+100